waiter/test/waiter/state_test.clj

```diff
@@ -1206,13 +1224,108 @@
                                               (map #(failed-instances-fn % (index-fn %)) expected-services))
                                       :service-id->deployment-error
                                       (into {} (filter second (zipmap expected-services
-                                                                      (map #(deployment-error-fn % (index-fn %)) expected-services))))})
+                                                                      (map #(deployment-error-fn % (index-fn %)) expected-services))))
+                                      :service-id->instability-issue {}})
+
                     state (async/<!! router-state-push-chan)
                     actual-state (dissoc state :iteration :service-id->healthy-instances :service-id->expired-instances :service-id->starting-instances
                                           :service-id->instance-counts :service-id->my-instance->slots :routers :time)]
                 (when (not= expected-state actual-state)
                   (clojure.pprint/pprint (clojure.data/diff expected-state actual-state)))
                 (is (= expected-state actual-state) (str (clojure.data/diff expected-state actual-state))))))
+          (async/>!! exit-chan :exit)))))
+  (testing "router-state-maintainer-deployment-errors-updated"
```

Comment:
Is this related to your change? It seems to be about deployment errors.

Reply:
The function got copied over while rebasing/merging.